ava-test/chmod.js

```diff
@@ -0,0 +1,356 @@
+});
+
+test('u+wx', t => {
+  if (process.platform !== 'win32') {
+    let result = shell.chmod('000', `${t.context.tmp}/chmod/file1`);
+    t.is(result.code, 0);
+    result = shell.chmod('u+wx', `${t.context.tmp}/chmod/file1`);
+    t.is(result.code, 0);
+    t.is(
+      fs.statSync(`${t.context.tmp}/chmod/file1`).mode & parseInt('300', 8),
+      parseInt('300', 8)
+    );
+    result = shell.chmod('644', `${t.context.tmp}/chmod/file1`);
+    t.is(result.code, 0);
+  }
+});
+
+test('Multiple symbolic modes at once', t => {
+  if (process.platform !== 'win32') {
+    let result = shell.chmod('000', `${t.context.tmp}/chmod/file1`);
+    t.is(result.code, 0);
+    result = shell.chmod('u+r,g+w,o+x', `${t.context.tmp}/chmod/file1`);
+    t.is(result.code, 0);
+    t.is(
+      fs.statSync(`${t.context.tmp}/chmod/file1`).mode & parseInt('421', 8),
+      parseInt('421', 8)
+    );
+    result = shell.chmod('644', `${t.context.tmp}/chmod/file1`);
+    t.is(result.code, 0);
+  }
+});
+
+test('u+rw,g+wx', t => {
+  if (process.platform !== 'win32') {
+    let result = shell.chmod('000', `${t.context.tmp}/chmod/file1`);
+    t.is(result.code, 0);
+    result = shell.chmod('u+rw,g+wx', `${t.context.tmp}/chmod/file1`);
+    t.is(result.code, 0);
+    t.is(
+      fs.statSync(`${t.context.tmp}/chmod/file1`).mode & parseInt('630', 8),
+      parseInt('630', 8)
+    );
+    result = shell.chmod('644', `${t.context.tmp}/chmod/file1`);
+    t.is(result.code, 0);
+  }
+});
+
+test('u-x,g+rw', t => {
+  let result = shell.chmod('700', `${t.context.tmp}/chmod/file1`);
+  t.is(result.code, 0);
+  result = shell.chmod('u-x,g+rw', `${t.context.tmp}/chmod/file1`);
+  t.is(result.code, 0);
+  t.is(
+    fs.statSync(`${t.context.tmp}/chmod/file1`).mode & parseInt('660', 8),
+    parseInt('660', 8)
+  );
+  result = shell.chmod('644', `${t.context.tmp}/chmod/file1`);
+  t.is(result.code, 0);
+});
+
+test('a-rwx,u+rw', t => {
+  let result = shell.chmod('a-rwx,u+rw', `${t.context.tmp}/chmod/file1`);
+  t.is(result.code, 0);
+  t.is(
+    fs.statSync(`${t.context.tmp}/chmod/file1`).mode & parseInt('600', 8),
+    parseInt('600', 8)
+  );
+  result = shell.chmod('a-rwx,u+rw', `${t.context.tmp}/chmod/file1`);
+  t.is(result.code, 0);
+  t.is(
+    fs.statSync(`${t.context.tmp}/chmod/file1`).mode & parseInt('600', 8),
+    parseInt('600', 8)
+  );
+  result = shell.chmod('644', `${t.context.tmp}/chmod/file1`);
+  t.is(result.code, 0);
+});
+
+test('Numeric modes', t => {
+  let result = shell.chmod('744', `${t.context.tmp}/chmod/xdir`);
+  t.is(result.code, 0);
+  result = shell.chmod('644', `${t.context.tmp}/chmod/xdir/file`);
+  t.is(result.code, 0);
+  result = shell.chmod('744', `${t.context.tmp}/chmod/xdir/deep`);
+  t.is(result.code, 0);
+  result = shell.chmod('644', `${t.context.tmp}/chmod/xdir/deep/file`);
+  t.is(result.code, 0);
+  result = shell.chmod('-R', 'a+X', `${t.context.tmp}/chmod/xdir`);
+  t.is(result.code, 0);
+});
+
+test('No Test Title #27', t => {
```

Comment:
I think this was actually a cleanup step before. There's no harm in running these lines and making sure they succeed though, so I kept them.